### PR TITLE
WIP: Add post_create_track method

### PR DESCRIPTION
**Pull request by abbiesims**
## Done
Adds method to call create_track endpoint in StoreAPI repo ([StoreAPI PR](https://github.com/canonical/canonicalwebteam.store-api/pull/114)) 

## QA
I tested this endpoint with https://snapcraft.io/fran-snap/listing as I have permissions to do so. Please check attached images to verify that:
- tracks can be created (with `v\\.[0-9]+` guardrail)
- a track that already exists returns the expected response
- a track that does not conform to the guardrails returns an invalid response

## Issue / Card
Fixes https://warthogs.atlassian.net/browse/WD-9842
Screenshot from 2024-03-28 17-18-40
Screenshot from 2024-03-28 17-19-16
Screenshot from 2024-03-28 17-19-28
Screenshot from 2024-03-28 17-19-41

### CODE REVIEW

File: webapp/publisher/views.py
Comment (steverydz) [inline webapp/publisher/views.py:1]: Can you  revert the change on this file?

**Review by steverydz — APPROVED** [review]
LGTM 👍 
Just one small comment in the code

File: webapp/publisher/views.py
Comment (abbiesims) [reply webapp/publisher/views.py:1]: Sure, done!
